xds/internal/balancer/edsbalancer/eds_testutil.go

```diff
@@ -111,3 +115,78 @@ func parseEndpoints(lbEndpoints []*endpointpb.LbEndpoint) []xdsclient.Endpoint {
 	}
 	return endpoints
 }
+
+// testPickerFromCh receives pickers from the channel, and check if their
+// behaviors are as expected (that the given function returns nil err).
+//
+// It returns nil if one picker has the correct behavior.
+//
+// It returns error when there's no picker from channel after 1 second timeout,
+// and the error returned is the mismatch error from the previous picker.
+func testPickerFromCh(ch chan balancer.Picker, f func(balancer.Picker) error) error {
+	var (
+		p   balancer.Picker
+		err error
+	)
+	for {
+		select {
+		case p = <-ch:
+		case <-time.After(time.Second):
+			return fmt.Errorf("timeout waiting for picker with expected behavior, error from previous picker: %v", err)
+		}
+
+		err = f(p)
+		if err == nil {
+			return nil
+		}
```

Comment:
If this block of code is moved into the `case p := <-ch`, then we wont need to declare the variables up front.

Reply:
We need to err var to keep the error from the previous picker, so we have more information when error on timeout.